src/features/common/utils.ts

```diff
@@ -17,10 +18,23 @@ export function isArc32AppSpec(appSpec: Arc32AppSpec | Arc4AppSpec): appSpec is
   )
 }
 
-export function isArc4AppSpec(appSpec: Arc32AppSpec | Arc4AppSpec): appSpec is Arc4AppSpec {
+export function isArc4AppSpec(appSpec: AppSpec): appSpec is Arc4AppSpec {
   return (
     appSpec !== null &&
     typeof appSpec === 'object' &&
+    !('arcs' in appSpec) &&
+    'methods' in appSpec &&
+    Array.isArray(appSpec.methods) &&
+    appSpec.methods.length > 0 &&
+    typeof appSpec.methods[0] === 'object'
+  )
+}
+
+export function isArc56AppSpec(appSpec: AppSpec): appSpec is Arc56Contract {
```

Comment:
"We should also check that 56 is included in the arcs array here" applies here too.
I'm wondering if we can share this check?

Reply:
good idea! I updated it to call `isXAppSpec` in `jsonAsXAppSpec` functions.